### ISSUE TITLE
cannot flash firmware stuck at 0%

### ISSUE THREAD

**Issue by sgupta999** [itead/Sonoff_Devices_DIY_Tools#64]
I have sonoff minis pre-flashed with version 3.3.0
When I use the tool I can see the device and turn it on and off from the tool.
But when flashing firmware it is stuck at 0%. I made sure that the wifi that Mini is connected has access to the internet

**Comment by dougbaptista** [comment]
Also here

**Comment by sgupta999** [comment]
Just FYI - i think it has something to do with the latest Win 10, 1909 update -  had updated it too..
the way i could make it work

1. I had bunch of virtual and other network adaptors enabled - disable all of those except the one wifi that you are using

2. for the the log version of the tool worked more consistently than the non-log version

again the fixes i came up with suggest something weird with windows most likely

**Comment by JosepZ** [comment]
Same here.

**Comment by dougbaptista** [comment]
I used another computer (notebook) and got it.

1) Download DIY Tool
2) HotSpot with cellphone (pc didn't work)
3) Connect to a network called sonoffDiy (SSID) with a password = 20170618sn.
4) Run the tool_01DIY85 file (3.3.0)
5) Check if it detects any device, and try to turn it on and off by DIY mode tool
6) Flash Firmware - Import Firmware and Flash device list and press OK

Wait for progress to reach 100%. If not out of place, close the program, reopen, repeat the process from step 5.

**Comment by sameeraman** [comment]
Check the following:
1) Set the SSID of the Sonoff Mini to the Home wifi via the DIY network.
2) Make sure your laptop is using 2.4GHz wireless connection to connect to the home network or use a wired connection
3) Disable if there any other network connections in the laptop
4) Turn off windows firewall on the laptop temporarily.
5) Run the DIY tool with windows 7 compatibility mood.
![image](https://user-images.githubusercontent.com/18417281/72196833-07bed480-3456-11ea-9a31-53370e55e151.png)

These things did the trick for me.

**Comment by JosepZ** [comment]
Worked. THANKS!

**Comment by vsailo** [comment]
Please ensure that your Sonoff hotspot device has access to Internet. After my various attempts I could get working as soon as I switched on my Mobile (Sonoff hotspot device) data.

**Comment by ErnieFR** [comment]
Same here with Win7, no firewall, Huawei Y5ii phone as hotspot (also with internet access (mobile data = on), not moving on once flashing starts. Always in DIY status info 'otaUnlock:false'. 
Thanks to "Vsailo" topic above, I used my main internet router's wifi. First attempt failed, now asking info in DIYtool suddenly shows "otoUnlock:true"! YES  Tried it again and it worked!
Now in my enthousiasm I forgot to check the IPaddress, but that is not my worry. After waiting for several minits the Sonoff only blinks regularly on/off no matter in DIY mode or not. 
`<b><center>`have I ruined my sonoff??  `</center></b>`

![DIYtool info with otaUnlock-true](https://user-images.githubusercontent.com/60966258/74351161-09621c00-4db7-11ea-8bf3-aaa10101da9b.JPG)

![DIYtool suddenly loads firmware](https://user-images.githubusercontent.com/60966258/74351269-33b3d980-4db7-11ea-8380-eae35a6d52f0.JPG)

**Comment by parad0xnasha** [comment]
I've tried everything listed here but no matter what i cannot progress past 0% as per the OP's issue and others mentioned here and elsewhere. Tried both Phone AP, laptop AP, 2.4GHz only network, firewall disabled, Win7 compatibility mode. Nothing works :(

**Comment by Taomyn** [comment]
I have the same problem as @parad0xnasha no matter what I have tried I can see the R3 and control and get data from it, but the flash never goes further than 0%

I fixed mine, it was the Windows firewall, when I switched from using the Windows hotspot to one on my phone, it set the network profile to Public that enables the firewall. Set it to Private that switches it of, restarted the R3 and and tool and it flashed first time. Hope this helps.

**Comment by ErnieFR** [comment]
> 
> 
> I've tried everything listed here but no matter what i cannot progress past 0% as per the OP's issue and others mentioned here and elsewhere. Tried both Phone AP, laptop AP, 2.4GHz only network, firewall disabled, Win7 compatibility mode. Nothing works :(

@parad0xnasha did you make sure the AP network had a way of connecting to internet at the same time ?

**Comment by parad0xnasha** [comment]
> > I've tried everything listed here but no matter what i cannot progress past 0% as per the OP's issue and others mentioned here and elsewhere. Tried both Phone AP, laptop AP, 2.4GHz only network, firewall disabled, Win7 compatibility mode. Nothing works :(
> 
> @parad0xnasha did you make sure the AP network had a way of connecting to internet at the same time ?

@ErnieFR - Yes, phone Hotspot utilises 3/4G connection to connect to the internet, and my laptop is connected to my home WiFi when using it as a hotspot

@Taomyn - I wish this solved my problem. I created exceptions in the firewall initially, but just for the sake of it to ensure it wasn't incorrectly configured i also disabled the firewall during the course of my testing.

**Comment by ErnieFR** [comment]
@parad0xnasha I think you're left with one last option and that is to connect it via usb-ftdi adapter

**Comment by parad0xnasha** [comment]
@ErnieFR Yes, i literally just purcchased an FTDI adapter off eBay (I've also contacted Itead support, because this is a widespread issue with DIY mode

**Comment by PeterWagemans** [comment]
Having the AP connect to internet solved my issue. WHY does it have to be able to access internet to flash a local file to a local device?

**Comment by ErnieFR** [comment]
> 
> 
> Having the AP connect to internet solved my issue. WHY does it have to be able to access internet to flash a local file to a local device?

I know, it totally makes no sense. Apparently there's more happening than meets the eye ....

**Comment by vkeiper** [comment]
> 
> 
> Just FYI - i think it has something to do with the latest Win 10, 1909 update - had updated it too..
> the way i could make it work
> 
>     1. I had bunch of virtual and other network adaptors enabled - disable all of those except the one wifi that you are using
> 
>     2. for the the log version of the tool worked more consistently than the non-log version
> 
> 
> again the fixes i came up with suggest something weird with windows most likely
@sgupta999,
thank you. I have been killing myself trying to get this mini programmed but kept getting stuck at 0%. After I disabled all my other Ethernet adapters like you mention. BOOM 1st shot.
